### PR TITLE
[WIP] Add filter categories to MAX and DAX assets

### PR DESCRIPTION
**Pull request by ckadner**
**Changes in this PR:**
* Add `filter_categories` to DAX datasets and MAX models
  ```YAML
  filter_categories:
    domain: "image-to-text"
    platform: 
      - "kubernetes"
      - "kfserving"
    language: "python"
    framework: "tensorflow"
  ```

**Related issues:**
https://github.com/machine-learning-exchange/mlx/pull/167
https://github.com/machine-learning-exchange/mlx/pull/183

### CODE REVIEW

**Comment by ckadner** [comment]
These are the filter categories I see for **MAX models**:
https://developer.ibm.com/exchanges/models/all/


**Models**

- Trainable
- Deployable

**Architectures & Deployment models**

- Serverless

**Model Asset Technologies**

- Image Classification
- Audio Classification
- Audio Feature Extraction
- Audio Modeling
- Image Feature Extraction
- Image-to-Image Translation or Transformation
- Image-to-Text Translation
- Language Modeling
- Show more

**Products & Services**

- Docker
- Keras
- MAX - Model Asset eXchange
- PyTorch
- TensorFlow

**Technologies**

- Artificial intelligence
- Deep learning
- Machine learning
- Natural language processing
- Speech and empathy
- Visual recognition

**Comment by ckadner** [comment]
These are the filter categories I find for **DAX Datasets**
https://developer.ibm.com/exchanges/data/all/


**Data Asset Technologies**

- Audio
- Classification
- Computational Argumentation
- Debater
- DevOps
- Document Layout Analysis
- Feature Extraction
- Image
- Show more

**Products & Services**

- DAX - Data Asset eXchange

**Technologies**

- Artificial intelligence

**Comment by ckadner** [comment]
Both the MAX and DAX filter categories need to be refined as they seem to be somewhat overlapping or vague. The filter categories for https://huggingface.co/datasets are much more concise, like `task`, `language`, `size`, `license`,

**Comment by ckadner** [comment]
@kmh4321

**Comment by ckadner** [comment]
We should create an authoritative list of `filter_categories` that users who upload models/datasets can rely on and expand on. 

The authoritative list will also be helpful when deciding which labels the UI will put on the featured cards (@drewbutlerbb4 ):

```YAML
    filter_categories:
      domain:      "image-to-text"
      platform:    ["kubernetes", "kfserving"]
      language:    "python"
      framework:   "tensorflow"
      industry:    "finance"
      application: ["devops", "CI/CD"]
      data:        ["audio", "video", "text"]
```

If we run out of good ideas for some more interesting categorizations, we could that a look at the LF conference submissions, where they slot talks into various categories (see https://linuxfoundation.smapply.io/prog/)
